### PR TITLE
OkHttpClient

### PR DESCRIPTION
**Pull request by javipacheco**
This PR use the `OkHttpClient` for API calls instead of the previous approach in order to integrate the new API votes

@fedefernandez can you please review? thanks

### CODE REVIEW

File: src/main/scala/com/fortysevendeg/android/scaladays/modules/net/client/ServiceClient.scala [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      }
+  }
+
+  private def transformResponse[T](
+    content: String,
+    reads: Reads[T]
+    ): Option[T] =
+    Try(Json.parse(content).as[T](reads)) match {
+      case Success(s) => Some(s)
+      case Failure(e) => None
```
Comment (fedefernandez) [inline src/main/scala/com/fortysevendeg/android/scaladays/modules/net/client/ServiceClient.scala]: Maybe we should call to `e.printStackTrace` here before return `None`

File: src/main/scala/com/fortysevendeg/android/scaladays/modules/net/client/messages/Messages.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+case class ServiceClientRequest[Res](
+  path: String,
+  headers: Seq[(String, String)] = Seq.empty,
+  reads: Option[Reads[Res]] = None,
+  emptyResponse: Boolean = false)
+
+case class ServiceClientWithBodyRequest[Res](
+  path: String,
+  headers: Seq[(String, String)] = Seq.empty,
+  body: Map[String, String],
```
Comment (fedefernandez) [inline src/main/scala/com/fortysevendeg/android/scaladays/modules/net/client/messages/Messages.scala]: We should add a default value for `body` param or move `header` param bellow `body`

File: src/main/scala/com/fortysevendeg/android/scaladays/modules/net/impl/NetServicesComponentImpl.scala [hunk 3]
```diff
@@ -43,21 +49,29 @@ trait NetServicesComponentImpl
...
+              case Some(true) => NetResponse(success = true, downloaded = true)
+              case _ => NetResponse(success = false, downloaded = false)
+            }
           }
-        } else {
-          NetResponse(success = true, downloaded = false)
+        } yield result
+        future.recoverWith {
+          case NonFatal(e) =>
+            // log
```
Comment (fedefernandez) [inline src/main/scala/com/fortysevendeg/android/scaladays/modules/net/impl/NetServicesComponentImpl.scala]: `android.util.Log.e("Error saving JSON", e)` ???

**Comment by fedefernandez** [comment]
Minor comments, besides that, LGTM!

File: src/main/scala/com/fortysevendeg/android/scaladays/modules/net/client/messages/Messages.scala (hunk 2 above)
Comment (javipacheco) [inline src/main/scala/com/fortysevendeg/android/scaladays/modules/net/client/messages/Messages.scala]: :+1:

**Pull request merged**